Tests/UnitTests/ExperienceEventTests.swift

```diff
@@ -124,6 +125,60 @@ class ExperienceEventTests: XCTestCase {
         XCTAssertTrue(NSDictionary(dictionary: actualEventData).isEqual(to: expectedEventData))
     }
 
+    func testAsDictionary_withXdmSchemaAndDataAndDatastreamIdOverride() {
+
+        let expectedData = generateData()
+        let expectedConfig: [String: Any] = ["datastreamIdOverride": "testDatastreamId"]
+        var expectedEventData: [String: Any] = [:]
+        expectedEventData[xdm] = expectedXdmSchema
+        expectedEventData[data] = expectedData
+        expectedEventData[datasetId] = "5dd603781b95cc18a83d42ce"
+        expectedEventData[config] = expectedConfig
+
+        let experienceEvent = ExperienceEvent(xdm: generatedXdmSchema, data: expectedData, datastreamIdOverride: "testDatastreamId")
+        guard let actualEventData = experienceEvent.asDictionary() else {
+            XCTFail("Failed to retrieve experience event asDictionary")
+            return
+        }
+        XCTAssertTrue(NSDictionary(dictionary: actualEventData).isEqual(to: expectedEventData))
+    }
+
+    func testAsDictionary_withXdmSchemaAndDataAndDatastreamConfigOverride() {
```

Comment:
can we also test with XdmSchema with empty dataset which should be the expected flow if you use datastream overrides